azkaban-common/src/main/java/azkaban/executor/ExecutorHealthChecker.java

```diff
@@ -169,11 +179,32 @@ private void handleExecutorNotAliveCase(
         (executor.getId(), 0) + 1);
     if (this.executorFailureCount.get(executor.getId()) % this.executorMaxFailureCount == 0
         && !this.alertEmails.isEmpty()) {
-      entry.getValue().stream().forEach(flow -> flow
-          .getExecutionOptions().setFailureEmails(this.alertEmails));
       logger.info(String.format("Executor failure count is %d. Sending alert emails to %s.",
           this.executorFailureCount.get(executor.getId()), this.alertEmails));
-      this.alerterHolder.get("email").alertOnFailedUpdate(executor, entry.getValue(), e);
+      this.alerterHolder.get("email").alertOnFailedExecutorHealthCheck(executor, entry.getValue(), e,
+          this.alertEmails);
+      this.cleanupForMissingExecutor(executor, entry.getValue());
```

Comment:
The conditional indicates that if no alert emails are configured then `cleanupForMissingExecutor` will not happen. Is that what we want?

Reply:
My initial thought was that we should not perform a cleanup silently without notifying any admins.
But on a closer inspection it makes sense to do so becuase:
- The corresponding message is being logged
- Flow failure emails will still go out to flow admins
- Keeps the behavior more consistent and predictable with or without admin email configured.
Thanks for pointing this out. Will make the change.

Reply:
Fixed.